src/api/controllers/user-controllers.ts

```diff
@@ -1,135 +1,94 @@
     try {
-        // Call to service layer
+        /** Call to service layer */
         const result = await userService.createSampleUser();
 
-        // Return a response to client.
+        /** Return a response to client. */
         return res.status(200).json(result);
-    } catch (error) {
-        const err = error as Error;
-        logging.error(err.message);
-
-        // Return a response to client.
-        return res.status(500).json({
-            error: error
-        });
+    } catch (err) {
+        next(err);
     }
 };
 
-const createSimpleUser = async (req: Request, res: Response) => {
-    logging.info(`${NAMESPACE}: Creating a simple user with request body`);
+const createSimpleUser = async (req: Request, res: Response, next: NextFunction) => {
     const user = req.body;
+    // Check for required request body
+    if (!user.firstName) {
+        const errMessage = 'Missing firstname.';
+        throw new APIError(errMessage, 'createSimpleUser', httpStatusCode.BAD_REQUEST);
+    }
+    if (!user.lastName) {
+        const errMessage = 'Missing lastname.';
+        throw new APIError(errMessage, 'createSimpleUser', httpStatusCode.BAD_REQUEST);
+    }
+    if (!user.email) {
+        const errMessage = 'Missing email.';
+        throw new APIError(errMessage, 'createSimpleUser', httpStatusCode.BAD_REQUEST);
+    }
     try {
-        // Check for required request body
-        if (!user.firstName) throw new MissingAttributeError('fileName', 'createSimpleUser');
-        if (!user.lastName) throw new MissingAttributeError('lastName', 'createSimpleUser');
-        if (!user.email) throw new MissingAttributeError('email', 'createSimpleUser');
-
-        // Call service layer
+        /** Call to service layer */
         const result = await userService.createSimpleUser(user);
 
-        // Return result
+        /** Return a response to client. */
         return res.status(200).json(result);
-    } catch (e) {
-        // Return error caught during check or service layer call
-        if (e instanceof Error) {
-            logging.error(e.message);
-            return res.status(400).json({ error: e });
-        }
-        return res.status(500).json({ error: e });
+    } catch (err) {
+        next(err);
     }
 };
 
-const getUserByEmail = async (req: Request, res: Response) => {
-    logging.info(`${NAMESPACE}: Getting the User with matching email`);
+const getUserByEmail = async (req: Request, res: Response, next: NextFunction) => {
     const email = req.body.email;
+    // Check request body
+    if (!email) throw new APIError('Missing Email.', 'getUserByEmail', httpStatusCode.BAD_REQUEST, true); //  Email presence check
+    // Regex match for email
+    if (!email.match(emailRegex)) throw new APIError('Where is the email, Lebowski?', 'getUserByEmail', httpStatusCode.BAD_REQUEST, true); //  Email validity check
+    // Return result
```

Comment:
Line 63. Err message should be about invalid email. Use that err message for line 61 instead.

Reply:
I agree